### DISCUSSION TITLE
How to echo something in a vimscript plugin?

### DISCUSSION THREAD

**Discussion by otakutyrant** [neovim/neovim#24366] [Q&A] [answered]
I inserted these lines into `sonokai/colors/sonokai-andromeda.vim`:

```
let basename = expand("%:t:r") " Like sonokai-default.
echo basename
```

After modification, I installed the sonokai plugin locally. But when I execute `colorscheme sonokai-andromeda`, no messages hinting the basename.

**Comment by zeertzjq** [comment]
Where did you insert these lines? You may have inserted them at an unreachable position.

**Reply by otakutyrant** [reply]
Like this: https://github.com/otakutyrant/sonokai/blob/multiple_colorschemes/colors/sonokai-andromeda.vim

**Reply by zeertzjq** [reply]
Hmm, that's reachable, but maybe something else clears the message. Try changing `:echo` to `:echomsg` and running `:messages` after the colorscheme is loaded.

**Reply by otakutyrant** [reply]
I found out that `echo "Hello"` or `echomsg "Hello"` work, but the message-history does not include any result of `echo basename`. Any idea?

**Comment by otakutyrant** [comment, marked as answer]
Blame vimscript!
